lib/ansible/modules/database/postgresql/postgresql_publication.py

```diff
@@ -0,0 +1,653 @@
+        Returns:
+            True if successful, False otherwise.
+        """
+        if check_mode:
+            self.executed_queries.append(query)
+            return True
+        else:
+            return exec_sql(self, query, ddl=True)
+
+
+# ===========================================
+# Module execution.
+#
+
+
+def main():
+    argument_spec = postgres_common_argument_spec()
+    argument_spec.update(
+        name=dict(required=True),
+        db=dict(type='str', aliases=['login_db']),
+        state=dict(type='str', default='present', choices=['absent', 'present']),
+        tables=dict(type='list'),
+        parameters=dict(type='dict'),
+        owner=dict(type='str'),
+        cascade=dict(type='bool'),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    # Parameters handling:
+    name = module.params['name']
+    state = module.params['state']
+    tables = module.params['tables']
+    params = module.params['parameters']
+    owner = module.params['owner']
+    cascade = module.params['cascade']
+
+    if state == 'absent':
+        if tables:
+            module.warn('parameter "tables" is ignored when "state=absent"')
+        if params:
+            module.warn('parameter "parameters" is ignored when "state=absent"')
+        if owner:
+            module.warn('parameter "owner" is ignored when "state=absent"')
+
+    if state == 'present' and cascade:
+        module.warm('parameter "cascade" is ignored when "state=present"')
+
+    # Connect to DB and make cursor object:
+    conn_params = get_conn_params(module, module.params)
+    # We check publication state without DML queries execution, so set autocommit:
+    db_connection = connect_to_db(module, conn_params, autocommit=True)
+    cursor = db_connection.cursor(cursor_factory=DictCursor)
+
+    # Check version:
+    if cursor.connection.server_version < SUPPORTED_PG_VERSION:
+        module.fail_json(msg="PostgreSQL server version should be 10.0 or greater")
+
+    # Nothing was changed by default:
+    changed = False
+
+    ###################################
+    # Create object and do rock'n'roll:
+    publication = PgPublication(module, cursor, name)
+
+    if tables:
+        tables = transform_tables_representation(tables)
+
+    # If module.check_mode=True, nothing will be changed:
+    if state == 'present':
+        if not publication.exists:
+            changed = publication.create(tables, params, owner, check_mode=module.check_mode)
+
+        else:
+            changed = publication.update(tables, params, owner, check_mode=module.check_mode)
+
+    elif state == 'absent':
+        changed = publication.drop(cascade=cascade, check_mode=module.check_mode)
+
+    # Get final publication info:
+    pub_fin_info = {}
+    if state != 'absent' or (state == 'absent' and module.check_mode):
+        pub_fin_info = publication.get_info()
+    elif state == 'absent' and not module.check_mode:
```

Comment:
Why not also do this when `module.check_mode` is `True`? Then the check mode behavior is closer to the real thing.

Reply:
yes, i thought about it too. it's difficult to recall now why i exactly implemented it like this but i remember, in short, if it was made another way it didn't work that i expected. I tried several ways, the current way looks better to me

Reply:
i left a comment here, where it is now.. strange :)

Reply:
in short, i tried several ways to do it in another manner but it didn't work that i expected. However it is only my opinion, if in the future people want to see it in another format, i'll change it.